### PR TITLE
web: separate build and deploy, incorporate landing page

### PR DESCRIPTION
**Pull request by williballenthin**
Update CI to separate the building and deployment jobs, which will enable other sub-pages to be hosted on GitHub pages. Demonstrate this by adding a trivial "hello world" page that gets shown at https://mandiant.github.io/capa.

Move the capa explorer web interface to https://mandiant.github.io/capa/explorer

Example: https://williballenthin.github.io/capa/

### Checklist

- [x] No CHANGELOG update needed

- [x] No new tests needed

- [x] No documentation update needed

### CODE REVIEW

File: .github/workflows/web-deploy.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
-name: deploy Capa Explorer Web to Github Pages
+name: deploy web to GitHub Pages
 
 on:
-  # Runs on pushes targeting the webui branch
   push:
-    branches: [ master ]
+    branches: [ master, "wb/webui-actions-1" ]
```
Comment (williballenthin) [inline .github/workflows/web-deploy.yml:5]: this is the branch on my fork that i develop patches. since its not found under mandiant/capa, can we keep this around?

File: .gitmodules [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [submodule "rules"]
 	path = rules
-	url = ../capa-rules.git
+	url = ../../mandiant/capa-rules.git
 [submodule "tests/data"]
 	path = tests/data
-	url = ../capa-testfiles.git
+	url = ../../mandiant/capa-testfiles.git
```
Comment (williballenthin) [inline .gitmodules:6]: sorry, unrelated change:

when a user forks capa, git looks for the submodules to be found under the users account, rather than at github.com/mandiant/capa-testfiles (for example). this change makes git look to mandiant/capa-testfiles (for example) by default.

**Comment by s-ff** [comment]
> blocked by #2266

merging #2267, should hopefully resolve this

File: .gitmodules (hunk 2 above)
Comment (s-ff) [reply .gitmodules:6]: Thanks, this confused some time ago

**Review by mr-tz — APPROVED** [review]
